--- a/build.rs
+++ b/build.rs
@@ -1,8 +1,6 @@
 use cfg_aliases::cfg_aliases;
-use vergen_git2::{BuildBuilder, CargoBuilder, Emitter, Git2Builder, RustcBuilder, SysinfoBuilder};
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    // Setup cfg aliases
+fn main() {
     cfg_aliases! {
         // Platforms
         macos: { target_os = "macos" },
@@ -14,20 +12,4 @@
         smb_unix: { all(unix, feature = "smb") },
         smb_windows: { all(windows, feature = "smb") }
     }
-
-    let build = BuildBuilder::all_build()?;
-    let cargo = CargoBuilder::all_cargo()?;
-    let git2 = Git2Builder::all_git()?;
-    let rustc = RustcBuilder::all_rustc()?;
-    let si = SysinfoBuilder::all_sysinfo()?;
-
-    Emitter::default()
-        .add_instructions(&build)?
-        .add_instructions(&cargo)?
-        .add_instructions(&git2)?
-        .add_instructions(&rustc)?
-        .add_instructions(&si)?
-        .emit()?;
-
-    Ok(())
 }
