--- negcache.cc
+++ negcache.cc
@@ -78,7 +78,8 @@ void NegCache::updateStaleEntry(time_t now, negcache_t::iterator& entry, QType qtype)
   // We we look how old the entry is, and increase d_servedStale accordingly, taking care not to overflow
   const time_t howlong = std::max(static_cast<time_t>(1), now - entry->d_ttd);
   const uint32_t extension = std::max(1U, std::min(entry->d_orig_ttl, s_serveStaleExtensionPeriod));
-  entry->d_servedStale = std::min(entry->d_servedStale + 1 + howlong / extension, static_cast<time_t>(s_maxServedStaleExtensions));
+  const time_t servedStale = std::min(static_cast<time_t>(entry->d_servedStale) + 1 + howlong / static_cast<time_t>(extension), static_cast<time_t>(s_maxServedStaleExtensions));
+  entry->d_servedStale = static_cast<uint16_t>(servedStale);
   entry->d_ttd = now + std::min(entry->d_orig_ttl, s_serveStaleExtensionPeriod);
 
   if (qtype == QType::ENT) {
