--- src/protocol_serial.c	2026-05-24 22:54:16.104251138 +0200
+++ src/protocol_serial.c	2026-05-24 22:54:52.154250213 +0200
@@ -13,6 +13,27 @@
 
 #define CMD_SIZE(cmd) ( sizeof(cmd) - sizeof(command_common_t) )
 
+static uint16_t to_target_le16(uint16_t v)
+{
+#if defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+    return (uint16_t)(((v & 0x00FFU) << 8) | ((v & 0xFF00U) >> 8));
+#else
+    return v;
+#endif
+}
+
+static uint32_t to_target_le32(uint32_t v)
+{
+#if defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+    return ((v & 0x000000FFU) << 24) |
+           ((v & 0x0000FF00U) << 8) |
+           ((v & 0x00FF0000U) >> 8) |
+           ((v & 0xFF000000U) >> 24);
+#else
+    return v;
+#endif
+}
+
 static uint8_t compute_checksum(const uint8_t *data, uint32_t size)
 {
     uint8_t checksum = 0xEF;
@@ -68,14 +89,14 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = FLASH_BEGIN,
-            .size = CMD_SIZE(flash_begin_cmd) - (encryption ? 0 : sizeof(uint32_t)),
-            .checksum = 0
+            .size = to_target_le16((uint16_t)(CMD_SIZE(flash_begin_cmd) - (encryption ? 0 : sizeof(uint32_t)))),
+            .checksum = to_target_le32(0)
         },
-        .erase_size = erase_size,
-        .packet_count = blocks_to_write,
-        .packet_size = block_size,
-        .offset = offset,
-        .encrypted = 0
+        .erase_size = to_target_le32(erase_size),
+        .packet_count = to_target_le32(blocks_to_write),
+        .packet_size = to_target_le32(block_size),
+        .offset = to_target_le32(offset),
+        .encrypted = to_target_le32(0)
     };
 
     *seq_num = 0;
@@ -96,11 +117,11 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = FLASH_DATA,
-            .size = CMD_SIZE(data_cmd) + size,
-            .checksum = compute_checksum(data, size)
+            .size = to_target_le16((uint16_t)(CMD_SIZE(data_cmd) + size)),
+            .checksum = to_target_le32(compute_checksum(data, size))
         },
-        .data_size = size,
-        .sequence_number = (*seq_num)++,
+        .data_size = to_target_le32(size),
+        .sequence_number = to_target_le32((*seq_num)++),
     };
 
     const send_cmd_config cmd_config = {
@@ -121,10 +142,10 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = FLASH_END,
-            .size = CMD_SIZE(end_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(end_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .stay_in_loader = stay_in_loader
+        .stay_in_loader = to_target_le32(stay_in_loader)
     };
 
     const send_cmd_config cmd_config = {
@@ -148,14 +169,14 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = FLASH_DEFL_BEGIN,
-            .size = CMD_SIZE(flash_begin_cmd) - (encryption ? 0 : sizeof(uint32_t)),
-            .checksum = 0
+            .size = to_target_le16((uint16_t)(CMD_SIZE(flash_begin_cmd) - (encryption ? 0 : sizeof(uint32_t)))),
+            .checksum = to_target_le32(0)
         },
-        .erase_size = erase_size,
-        .packet_count = blocks_to_write,
-        .packet_size = block_size,
-        .offset = offset,
-        .encrypted = 0
+        .erase_size = to_target_le32(erase_size),
+        .packet_count = to_target_le32(blocks_to_write),
+        .packet_size = to_target_le32(block_size),
+        .offset = to_target_le32(offset),
+        .encrypted = to_target_le32(0)
     };
 
     *seq_num = 0;
@@ -176,11 +197,11 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = FLASH_DEFL_DATA,
-            .size = CMD_SIZE(data_cmd) + size,
-            .checksum = compute_checksum(data, size)
+            .size = to_target_le16((uint16_t)(CMD_SIZE(data_cmd) + size)),
+            .checksum = to_target_le32(compute_checksum(data, size))
         },
-        .data_size = size,
-        .sequence_number = (*seq_num)++,
+        .data_size = to_target_le32(size),
+        .sequence_number = to_target_le32((*seq_num)++),
     };
 
     const send_cmd_config cmd_config = {
@@ -201,10 +222,10 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = FLASH_DEFL_END,
-            .size = CMD_SIZE(end_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(end_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .stay_in_loader = stay_in_loader
+        .stay_in_loader = to_target_le32(stay_in_loader)
     };
 
     const send_cmd_config cmd_config = {
@@ -223,11 +244,11 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = READ_FLASH_ROM,
-            .size = CMD_SIZE(flash_read_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(flash_read_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .address = address,
-        .size = READ_FLASH_ROM_DATA_SIZE,
+        .address = to_target_le32(address),
+        .size = to_target_le32(READ_FLASH_ROM_DATA_SIZE),
     };
 
     const send_cmd_config cmd_config = {
@@ -249,13 +270,13 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = READ_FLASH_STUB,
-            .size = CMD_SIZE(flash_read_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(flash_read_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .address = address,
-        .total_size = size,
-        .packet_data_size = size_per_packet,
-        .max_inflight_packets = 1,
+        .address = to_target_le32(address),
+        .total_size = to_target_le32(size),
+        .packet_data_size = to_target_le32(size_per_packet),
+        .max_inflight_packets = to_target_le32(1),
     };
 
     const send_cmd_config cmd_config = {
@@ -274,8 +295,8 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = ERASE_FLASH,
-            .size = CMD_SIZE(erase_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(erase_cmd)),
+            .checksum = to_target_le32(0)
         },
     };
 
@@ -294,11 +315,11 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = ERASE_REGION,
-            .size = CMD_SIZE(erase_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(erase_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .offset = offset,
-        .size = size,
+        .offset = to_target_le32(offset),
+        .size = to_target_le32(size),
     };
 
     const send_cmd_config cmd_config = {
@@ -316,8 +337,8 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = SYNC,
-            .size = CMD_SIZE(sync_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(sync_cmd)),
+            .checksum = to_target_le32(0)
         },
         .sync_sequence = {
             0x07, 0x07, 0x12, 0x20,
@@ -344,13 +365,13 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = SPI_ATTACH,
-            .size = loader->_stub_running
+            .size = to_target_le16(loader->_stub_running
             ? CMD_SIZE(attach_cmd) - sizeof(attach_cmd.zero)
-            : CMD_SIZE(attach_cmd),
-            .checksum = 0
+            : CMD_SIZE(attach_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .configuration = config,
-        .zero = 0
+        .configuration = to_target_le32(config),
+        .zero = to_target_le32(0)
     };
 
     const send_cmd_config cmd_config = {
@@ -369,13 +390,13 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = SPI_FLASH_MD5,
-            .size = CMD_SIZE(md5_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(md5_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .address = address,
-        .size = size,
-        .reserved_0 = 0,
-        .reserved_1 = 0
+        .address = to_target_le32(address),
+        .size = to_target_le32(size),
+        .reserved_0 = to_target_le32(0),
+        .reserved_1 = to_target_le32(0)
     };
 
     const send_cmd_config cmd_config = {
@@ -396,15 +417,15 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = SPI_SET_PARAMS,
-            .size = 24,
-            .checksum = 0
+            .size = to_target_le16(24),
+            .checksum = to_target_le32(0)
         },
-        .id = 0,
-        .total_size = total_size,
-        .block_size = 64 * 1024,
-        .sector_size = 4 * 1024,
-        .page_size = 0x100,
-        .status_mask = 0xFFFF,
+        .id = to_target_le32(0),
+        .total_size = to_target_le32(total_size),
+        .block_size = to_target_le32(64 * 1024),
+        .sector_size = to_target_le32(4 * 1024),
+        .page_size = to_target_le32(0x100),
+        .status_mask = to_target_le32(0xFFFF),
     };
 
     const send_cmd_config cmd_config = {
@@ -423,13 +444,13 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = MEM_BEGIN,
-            .size = CMD_SIZE(mem_begin_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(mem_begin_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .total_size = size,
-        .blocks = blocks_to_write,
-        .block_size = block_size,
-        .offset = offset
+        .total_size = to_target_le32(size),
+        .blocks = to_target_le32(blocks_to_write),
+        .block_size = to_target_le32(block_size),
+        .offset = to_target_le32(offset)
     };
 
     *seq_num = 0;
@@ -450,11 +471,11 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = MEM_DATA,
-            .size = CMD_SIZE(data_cmd) + size,
-            .checksum = compute_checksum(data, size)
+            .size = to_target_le16((uint16_t)(CMD_SIZE(data_cmd) + size)),
+            .checksum = to_target_le32(compute_checksum(data, size))
         },
-        .data_size = size,
-        .sequence_number = (*seq_num)++,
+        .data_size = to_target_le32(size),
+        .sequence_number = to_target_le32((*seq_num)++),
     };
 
     const send_cmd_config cmd_config = {
@@ -475,10 +496,10 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = MEM_END,
-            .size = CMD_SIZE(end_cmd),
+            .size = to_target_le16(CMD_SIZE(end_cmd)),
         },
-        .stay_in_loader = (entrypoint == 0),
-        .entry_point_address = entrypoint
+        .stay_in_loader = to_target_le32(entrypoint == 0),
+        .entry_point_address = to_target_le32(entrypoint)
     };
 
     const send_cmd_config cmd_config = {
@@ -498,13 +519,13 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = WRITE_REG,
-            .size = CMD_SIZE(write_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(write_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .address = address,
-        .value = value,
-        .mask = mask,
-        .delay_us = delay_us
+        .address = to_target_le32(address),
+        .value = to_target_le32(value),
+        .mask = to_target_le32(mask),
+        .delay_us = to_target_le32(delay_us)
     };
 
     const send_cmd_config cmd_config = {
@@ -523,10 +544,10 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = READ_REG,
-            .size = CMD_SIZE(read_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(read_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .address = address,
+        .address = to_target_le32(address),
     };
 
     const send_cmd_config cmd_config = {
@@ -546,11 +567,11 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = CHANGE_BAUDRATE,
-            .size = CMD_SIZE(baudrate_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(baudrate_cmd)),
+            .checksum = to_target_le32(0)
         },
-        .new_baudrate = new_baudrate,
-        .old_baudrate = old_baudrate
+        .new_baudrate = to_target_le32(new_baudrate),
+        .old_baudrate = to_target_le32(old_baudrate)
     };
 
     const send_cmd_config cmd_config = {
@@ -571,8 +592,8 @@
         .common = {
             .direction = WRITE_DIRECTION,
             .command = GET_SECURITY_INFO,
-            .size = CMD_SIZE(get_security_info_cmd),
-            .checksum = 0
+            .size = to_target_le16(CMD_SIZE(get_security_info_cmd)),
+            .checksum = to_target_le32(0)
         },
     };
 
