--- numpy/_core/src/common/npstd.hpp.orig
+++ numpy/_core/src/common/npstd.hpp
@@ -3,8 +3,10 @@
 
 #include <numpy/npy_common.h>
 
+#include <features.h>
+
 #include <cstddef>
 #include <cstring>
 #include <cctype>
 #include <cstdint>
 
 #include <string>
--- numpy/_core/src/common/utils.hpp.orig
+++ numpy/_core/src/common/utils.hpp
@@ -7,8 +7,10 @@
     #include <bit>
 #endif
 
+#include <features.h>
+
 #include <type_traits>
 #include <string.h>
 #include <cstdint>
 #include <cassert>
 
--- numpy/_core/src/highway/hwy/abort.cc.orig
+++ numpy/_core/src/highway/hwy/abort.cc
@@ -6,9 +6,11 @@
 
 #include "hwy/abort.h"
 
+#include <features.h>
+
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
 
 #include <atomic>
 #include <string>
--- numpy/_core/src/highway/hwy/targets.h.orig
+++ numpy/_core/src/highway/hwy/targets.h
@@ -25,9 +25,15 @@
 // For SIMD module implementations and their callers. Defines which targets to
 // generate and call.
 
+#include <features.h>
+
 #include "hwy/base.h"
 #include "hwy/detect_targets.h"
 #include "hwy/highway_export.h"
 
+#ifdef __UCLIBC__
+#define HWY_NO_LIBCXX
+#endif
+
 #if !defined(HWY_NO_LIBCXX)
 #include <atomic>