--- a/src/third-party/notcurses/src/lib/metric.c
+++ b/src/third-party/notcurses/src/lib/metric.c
@@ -1,4 +1,9 @@
-#include <fenv.h>
+#if defined(__has_include)
+# if __has_include(<fenv.h>)
+#  include <fenv.h>
+#  define LNAV_HAVE_FENV_H 1
+# endif
+#endif
 #include <string.h>
 #include <locale.h>
 #include <pthread.h>
@@ -28,7 +33,9 @@
 const char* ncnmetric(uintmax_t val, size_t s, uintmax_t decimal,
                       char* buf, int omitdec, uintmax_t mult,
                       int uprefix){
-  fesetround(FE_TONEAREST); // FIXME global to the process...ick :/
+#ifdef LNAV_HAVE_FENV_H
+  fesetround(FE_TONEAREST); // FIXME global to the process...ick :/
+#endif
   // these two must have the same number of elements
   const wchar_t* subprefixes = SUBPREFIXES;
   const wchar_t prefixes[] = L"KMGTPEZY"; // 10^21-1 encompasses 2^64-1