--- a/src/third-party/rapidyaml/ryml_all.hpp.orig
+++ b/src/third-party/rapidyaml/ryml_all.hpp
@@ -421,7 +421,7 @@
        #define C4_CPU_ARM
        #define C4_WORDSIZE 4
        #if defined(__ARM_ARCH_8__) || defined(__ARM_ARCH_8A__)  \
-        || (defined(__ARCH_ARM) && __ARCH_ARM >= 8)
-        || (defined(__TARGET_ARCH_ARM) && __TARGET_ARCH_ARM >= 8)  \
+        || (defined(__ARCH_ARM) && __ARCH_ARM >= 8)  \
+        || (defined(__TARGET_ARCH_ARM) && __TARGET_ARCH_ARM >= 8)
            #define C4_CPU_ARMV8
        #elif defined(__ARM_ARCH_7__) || defined(_ARM_ARCH_7)    \
